### PR TITLE
Missiles Initial Commit

### PR DESCRIPTION
**Pull request by Anton-Kr**
Adds missiles
https://github.com/Baystation12/Baystation12/pull/25716 a lot of code pulled and highly modified from here. Thank you @halworsen .

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -31,7 +31,7 @@
 	var/obj/effect/flood/flood_object
 	var/fluid_blocked_dirs = 0
 	var/flooded // Whether or not this turf is absolutely flooded ie. a water source.
-	var/height = 0 
+	var/height = 0
```
Comment (Anton-Kr) [inline code/game/turfs/turf.dm:34]: Remove this

File: code/modules/overmap/contacts/contact_sensors.dm [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//Seriously whoever made this fucking mess fuck you, I fucking hate you and your unmodular incomphrehensible half working fucking shit
```
Comment (Anton-Kr) [inline code/modules/overmap/contacts/contact_sensors.dm:1]: ... and this

File: code/modules/overmap/projectiles/missiles/equipment/utility/cargo.dm [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Passenger compartment, for the very brave and stupid
```
Comment (Anton-Kr) [inline code/modules/overmap/projectiles/missiles/equipment/utility/cargo.dm]: This does not work... well, and only interacts with a non assembled missile component. Refactor.

File: code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm [hunk 4]
```diff
@@ -0,0 +1,100 @@
+// Passenger compartment, for the very brave and stupid
```
Comment (Anton-Kr) [inline code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm:1]: This allows you stupidly enough to carry multiple people in your backpack. Significant refactor needed.

File: code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
+// Lights up the overmap
```
Comment (Anton-Kr) [inline code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm]: not compatible with our overmap system. Refactor or remove.

File: code/modules/overmap/sectors.dm [hunk 6]
```diff
@@ -5,24 +5,18 @@
 	name = "map object"
 	scannable = TRUE
 
-	var/list/map_z = list()
```
Comment (Anton-Kr) [inline code/modules/overmap/sectors.dm:8]: uhhh why did I remove this again what the fuck

**Comment by Armyguy** [comment]
Let me know when this gets to a point I can slap it on the vox ship, even for testing.

**Comment by Anton-Kr** [comment]
@Armyguy Goal is to get this merged first, and to get everyone to make a map change to include missiles in their specific away site which we could sync merge. I'll ping you.

File: code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm (hunk 4 above)
Comment (Vizoplez) [reply code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm:1]: This shouldn't be the case unless missile_equipment is literally undefined in terms of size?

Realistically, it should be a semi-structure with it requiring to be dragged & dropped, or similar to CM.

People picking up entire missiles is just strange.

File: code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm (hunk 5 above)
Comment (Vizoplez) [reply code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm]: Just make it function like current sensor beacons, or refactor the overmap to support the new lighting system. Its probably less work to migrate our current sensor net to the new one.

**Comment by Anton-Kr** [comment]
After some thoughts - passenger and cargo missiles are niche as fuck and honestly can be better handled by OFD anyway. If someone wants to readd them, by all means, but I am not willing to spend a day doing it at this time. Sensors I'll take a look at tmrw but they might get the same rope.

File: code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm (hunk 4 above)
Comment (Anton-Kr) [reply code/modules/overmap/projectiles/missiles/equipment/utility/passenger.dm:1]: It's a missile part which is an object and isn't bulky which is fairly silly.

File: code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm (hunk 5 above)
Comment (Anton-Kr) [reply code/modules/overmap/projectiles/missiles/equipment/utility/sensor.dm]: Honestly... later... maybe.

**Comment by ProbablyCarl** [comment]
Test merging.

**Comment by Anton-Kr** [comment]
Following playtest a good amt of bugs discovered. Will continue to develop.

**Comment by Anton-Kr** [comment]
@ProbablyCarl assuming this passes CI, this is ready for second testmerge.

**Pull request merged**